frontend/projects/upgrade/src/app/core/segments/store/segments.effects.ts

```diff
@@ -55,6 +55,35 @@ export class SegmentsEffects {
     )
   );
 
+  fetchSegmentOnSearchString$ = createEffect(
+    () =>
+      this.actions$.pipe(
+        ofType(SegmentsActions.actionSetSearchString),
+        map((action) => action.searchString),
+        tap((searchString) => {
+          // Allow empty string as we erasing text from search input
+          if (searchString !== null) {
+            this.store$.dispatch(SegmentsActions.actionFetchSegments({ fromStarting: true }));
```

Comment:
We don't have any pagination for Segment so this fetch should not be triggered on change in the Search string